modules/programs/tarsnap.nix

```diff
@@ -0,0 +1,300 @@
+        default = null;
+        description = ''
+          Limit upload bandwidth used to maxbwRate bytes per second.
+        '';
+      };
+
+      nodump = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Limit upload bandwidth used to maxbwRate bytes per second.
+        '';
+      };
+
+      printStats = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Print statistics for the archive being created (c mode) or delete (d
+          mode).
+        '';
+      };
+
+      quiet = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Avoid printing some warnings. Currently the warnings which are
+          silenced by this option are "Removing leading '/' ...", "Not adding
+          cache directory to archive", "... file may have grown while being
+          archived", and "Skipping entry on filesystem of type ...", but it is
+          likely that other warnings will be silenced by this option in future
+          versions of <command>tarsnap</command>.
+        '';
+      };
+
+      retryForever = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          This option causes tarsnap to continue trying to reconnect to the
+          tarsnap server forever, instead of giving up after 5-10 minutes. This
+          may be useful for people with excessively flaky networks, or on mobile
+          devices which regularly lose their internet connections for extended
+          periods of time. This is not enabled by default since continued
+          failures generally indicate a problem which should be investigated
+          by the user.
+        '';
+      };
+
+      storeAtime = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the storing of file access times. The default behaviour of
+          tarsnap is to not store file access times, since this can cause a
+          significant amount of bandwidth and storage to be wasted when the same
+          set of files are archived several times (e.g., if daily backup
+          archives are created) due to tarsnap itself accessing files and
+          thereby causing their access times to be changed.
+        '';
+      };
+
+      totals = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Print the size of the archive after creating it. This option is
+          provided mainly for compatibility with GNU tar; in most situations the
+          printStats option will be far more useful.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    home.file.".tarsnaprc".text = with cfg;
+      concatStringsSep "\n" ([
+        "cachedir ${
+          replaceStrings [ "$HOME" ] [ config.home.homeDirectory ] cachedir
```

Comment:
I would suggest removing the cachedir option and hard coding this to something like `${config.xdg.cacheHome}/tarsnap`.

Reply:
:+1: a